mol-loader/rdkit_utils.py

```diff
@@ -90,3 +93,82 @@ def rdkit_mol_from_smiles(smiles):
         except ValueError:
             pass
     return mol
+
+
+def rdkit_fps_from_mol(mol):
+    """
+    Generate a string representation of the fingerprint of mol
+
+    :param mol: RDKit Molecule
+    :return: String representation of RDKit fingerprint of mol, number of bits in the fingerprint
+    """
+    fps = None
+    num_bits = None
+    if mol is not None:
+        try:
+            fp = Chem.RDKFingerprint(mol)
+            num_bits = fp.GetNumOnBits()
+            fps = DataStructs.BitVectToFPSText(fp)
+        except ValueError:
+            pass
+    return fps, num_bits
+
+
+def rdkit_minhash_signature(mol, width):
+    """
+    Generate a MinHash signature of the specified width from the given molecule
+
+    :param mol: RDKit Molecule
+    :param width: The number of items to include in the hash signature
+    :return: List of up to the specified width of integers
+    """
+    sig = None
+    if mol is not None:
+        try:
+            all_hashes = []
+            fp = Chem.RDKFingerprint(mol)
+            # For small molecules, this will result in too few hashes for a valid signature.
+            # Suggestion: hash groups of features, instead of individual features.
+            # See http://infolab.stanford.edu/~ullman/mmds/ch3.pdf
+            #     Page 113, Section 3.8.4, "Matching Fingerprints"
+            # Challenge: the following molecules have 0 bits set in their fingerprints:
```

Comment:
The "0 bit" problem happens because the RDKit fingerprint is based around bonds: molecules without bonds don't set any bits. If you switch to the Morgan fingerprint, which is based around atoms, this problem goes away.